graphs/possible_bipartition.py

```diff
@@ -1,12 +1,50 @@
 # Can be used for BFS
-from collections import deque 
+from collections import deque
+from dis import dis 
 
 def possible_bipartition(dislikes):
     """ Will return True or False if the given graph
         can be bipartitioned without neighboring nodes put
         into the same partition.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """
-    pass
+    if not dislikes:
+        return True
+
+    seen = {}
+    for dog in range(len(dislikes)):
+        seen[dog] = False
+
+    q = deque()
+    play_area1 = set()
+    play_area2 = set()
+
+    first_dog = None
+    for clash in dislikes:
+        if len(clash) > 0:
+            first_dog = clash[0]
+            break
```

Comment:
🤔 This will end up working so long as `clash[0]` has neighbors, but I think there's a slight misunderstanding here about what `dislikes` holds. 

The nodes (aka dogs) in `dislikes` are actually represented by the _indices_ of the elements. The _value_ at each index, is the list of neighbors (aka edges or dogs that dog [index] dislikes) that node [index] has. 

For example in the graph below:
`dislikes = [ [], [2,3], [1,3], [] ]`
Node 0 has no neighbors. 
Node 1 has an edge that points to nodes 2 and 3.
Node 2 has an edge that points to node 1 and an edge that points to node 3.
Node 3 has no edges/neighbors. 

In the above example, your code would place Node 3 as the starting node, but your while loop on line 33 would finish after one iteration because Node 3 doesn't have any neighbors. There are edges pointing _to_ Node 3, but no edges stemming _from_ Node 3. In terms of our problem this would be because although Dogs 1 and 2 dislike Dog 3, but Dog 3 doesn't dislike Dogs 1 & 2 back.

What you would actually want to append is the first node with neighbors - which is Node 1. 

(You could also imagine an edge case where you can't reach all nodes from the first node with neighbors. You could consider refactoring your code to account for this case as well).